azext_iot/_params.py

```diff
@@ -94,7 +96,7 @@ def load_arguments(self, _):
         context.argument("resource_group_name", arg_type=resource_group_name_type)
         context.argument(
             "hub_name", options_list=["--hub-name", "-n"], arg_type=hub_name_type,
-            help="IoT Hub name. Required if --login is not provided.",
+            help="IoT Hub name or hostname. Required if --login is not provided.",
```

Comment:
Take a look at how this is handled in [digital twins](https://github.com/Azure/azure-iot-cli-extension/blob/dev/azext_iot/digitaltwins/params.py#L51C14-L51C30), a new argument is used to fully differentiate from control plane only name argument.

Reply:
the extension should have only dataplane commands for dps. - so all dps_names -> dps_name_or_hostname

for hub, only message endpoint and state should be control plane (with state possibly having some more logic on determining if dataplane only)